db_sql.go

```diff
@@ -43,25 +44,42 @@ type sqlAuthenticationDB struct {
 	db *sql.DB
 }
 
-func (x *sqlAuthenticationDB) Authenticate(identity, password string) (id string, er error) {
-	row := x.db.QueryRow(`SELECT identity, password FROM authuser WHERE LOWER(identity) = $1`, CanonicalizeIdentity(identity))
-	dbHash := ""
-	id = identity
-	if err := row.Scan(&id, &dbHash); err != nil {
+func (x *sqlAuthenticationDB) Authenticate(identity, password string) (er error) {
+	//check identity exists
+	row := x.db.QueryRow(`SELECT userid FROM userstore WHERE LOWER(identity) = $1`, CanonicalizeIdentity(identity))
```

Comment:
Write this function in a linear fashion, the same as one of the earlier ones that I mentioned. On error, return, etc. No nesting on success.
Also - I don't think there's any need for the comment "check identity exists". It's pretty clear from "SELECT userid FROM userstore".